core/src/analyser/rules/proxies.rs

```diff
@@ -129,18 +80,18 @@ impl Index<usize> for SharedTensorsProxy {
 /// solver.equals(input.rank, 2)
 /// solver.equals(input.shape[1], output.value[0][1])
 /// ```
-pub struct SharedTensorProxy {
+pub struct TensorProxy {
```

Comment:
So, `SharedTensorProxy` is now being replaced with tvec/slice being used directly?

Reply:
There are two things here. 
First a bad name, coming from an over-enthusiast regexp. SharedTensorProxy should be TensorProxy.
Second, the SharedTensorsProxy is gone (with an S at tensors). The rationale being that the input and output arity is always known before we reach the inference stage, so we can work with a regular slice there. It simplifies a bit processing operators with big lists of arguments (like the classifier or the lstm, so i though it was time to do this).